lib/metadata/include/irods/metadata.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef IRODS_METADATA_HPP
+#define IRODS_METADATA_HPP
+
+#include <string>
+#include <vector>
+#include <map>
+
+#include "irods/entity.hpp"
+#include "irods/rcConnect.h"
```

Comment:
Would we be able to avoid this include if we forward declared `RsComm` and `RcComm`?

Reply:
If the types are pointers or references, then we can forward declare them.